File: python/examples/talker.py
#!/usr/bin/env python3
"""irap_noroslib talker -- publishes std_msgs/String at 10 Hz to a real roscore.

Run a roscore, then:   python3 talker.py
Watch it with real ROS: rostopic echo /chatter
"""
import os
import irap_noroslib
from irap_noroslib.std_msgs.msg import String


def main():
    # Point irap_noroslib at the ROS master before init_node (defaults to a local roscore).
    irap_noroslib.set_master_uri(os.environ.get("ROS_MASTER_URI", "http://localhost:11311"))
    irap_noroslib.set_hostname(os.environ.get("ROS_HOSTNAME", "localhost"))
    irap_noroslib.init_node("irap_noroslib_talker")
    pub = irap_noroslib.Publisher("/chatter", String)
    rate = irap_noroslib.Rate(10)
    i = 0
    while not irap_noroslib.is_shutdown():
        m = String(data="hello world %d" % i)
        pub.publish(m)
        irap_noroslib.loginfo("published: %s" % m.data)
        i += 1
        rate.sleep()


if __name__ == "__main__":
    main()

File: cpp/examples/talker.cpp
// irap_noroslib talker — publish std_msgs/String at 10 Hz to a real roscore.
//   rostopic echo /chatter        # watch it from real ROS
#include "irap_noroslib.hpp"
#include <cstdlib>

int main() {
  // Point irap_noroslib at the ROS master before init_node (defaults to a local roscore).
  const char* mu = std::getenv("ROS_MASTER_URI");
  const char* hn = std::getenv("ROS_HOSTNAME");
  irap_noroslib::set_master_uri(mu ? mu : "http://localhost:11311");
  irap_noroslib::set_hostname(hn ? hn : "localhost");
  irap_noroslib::init_node("irap_noroslib_talker");
  irap_noroslib::Publisher<std_msgs::String> pub("/chatter");
  irap_noroslib::Rate rate(10);
  int i = 0;
  while (irap_noroslib::ok()) {
    std_msgs::String m;
    m.data = "hello world " + std::to_string(i++);
    pub.publish(m);
    irap_noroslib::loginfo("published: " + m.data);
    rate.sleep();
  }
  irap_noroslib::shutdown("done");
  return 0;
}
